akka-cluster-sharding-typed/src/main/scala/akka/cluster/sharding/typed/javadsl/ClusterSharding.scala

```diff
@@ -350,7 +350,7 @@ object StartEntity {
    * Returns [[ShardingEnvelope]] which can be sent via Cluster Sharding in order to wake up the
    * specified (by `entityId`) Sharded Entity, ''without'' delivering a real message to it.
    */
-  def create[M](msgClass: Class[M], entityId: String): ShardingEnvelope[M] =
+  def create[M](entityId: String): ShardingEnvelope[M] =
```

Comment:
I couldn't see why we need a msgClass here, any idea?

Reply:
I suspect it might be because where `javadsl.StartEntity.create(classOf[java.lang.Integer], "def")` created a `ShardingEnvelope<M>`, `javadsl.StartEntity.create("def")` creates a `ShardingEnvelope<Object>`, and it's not super obvious you'd need to write `javadsl.StartEntity.<Int>create("def")`? I might be wrong though - in any case we should have/write example code to demonstrate this.

Reply:
yes, the reason is to be explicit about the `M` type

note that we shouldn't change API in 2.5.x, so this change can't be merged to current master

Reply:
I'll revert this change then :+1: